src/src_main/crypto_tc.c

```diff
@@ -891,13 +890,12 @@ int32_t Crypto_TC_ProcessSecurity(uint8_t* ingest, int *len_ingest, TC_t* tc_sdl
     {
         // TODO - implement non-AEAD algorithm logic
 
-        if(sa_service_type == SA_ENCRYPTION || sa_service_type == SA_AUTHENTICATED_ENCRYPTION)
-        { 
+        if(sa_service_type == SA_ENCRYPTION) 
+        {
             status = cryptography_if->cryptography_decrypt();
         }
-        if(sa_service_type == SA_AUTHENTICATION || sa_service_type == SA_AUTHENTICATED_ENCRYPTION)
+        if(sa_service_type == SA_AUTHENTICATION)
```

Comment:
Same here, EST=1 & AST=1 is valid here still -- it'll be a non-AEAD algorithm using both decryption & validate authentication. Even though we don't support it yet, the condition was still correct and should be restored.

Reply:
In fact, now that I'm thinking about this -- the order is wrong. Shouldn't it authenticate before it decrypts in the authenticated encryption non-AEAD scenario? These calls need to be reversed. (Doesn't matter yet since we don't support any non-AEAD encryption algos yet)

Reply:
You are correct - ahead of you on that one. :)